### PR TITLE
documentation: update extension documentation

### PR DESCRIPTION
**Pull request by vince-fugnitto**
[**Rendered Readme**](https://github.com/vince-fugnitto/local-history-ext/blob/802f37997554824dcd95d0fc36aa47d617c56400/README.md)

**Todo**

- [x] go over the command documentation
- [x] go over the preference documentation
- [x] add documentation regarding the `Local-History View`

**Description**

The following commit:

- updates `maxEntriesPerFile` to default to **0** (display all revisions)
- adds documentation for high-level features
- adds documentation for the available commands
- fixes some formatting issues with the readme
- includes screenshots of the extension
- documents the available preferences

Signed-off-by: vince-fugnitto <vincent.fugnitto@ericsson.com>

### CODE REVIEW

**Comment by vince-fugnitto** [comment]
Please let me know if you guys have any additional suggestions or feedback :smiley:

**Comment by kaiyue0329** [comment]
@vince-fugnitto Should we mention that revisions are `read-only`?

I am thinking if we should include a screen recording of saving a file and seeing the revision show up on in the tree.

**Comment by vince-fugnitto** [comment]
> @vince-fugnitto Should we mention that revisions are `read-only`?
> 
> Also, I am thinking if we should include a screen recording of saving a file and seeing the revision show up on in the tree.

@kaiyue0329 sure, where would you propose we talk about `readonly` revisions? in the features?
Yes we will likely need to talk about the view as well in the features with screenshots.

File: README.md [hunk 1]
```diff
@@ -3,26 +3,130 @@
...
+
+The **Local History** extension manages local revisions of a file independently of source control providers, and allows users to easily view, track and restore files based on previous saves. The extension automatically creates save backups for the user so they are able to easily recover past revisions which is especially useful for disaster recovery.
+
+## Features
+
+The following are some high-level features the extension provides:
+
+- ### Automatic Revisions
+
+  The extension creates automatic revisions for the user based on when files are saved in their workspace. These automatic revisions are tracked by the extension and users are capable or reviewing them directly in the application.
```
Comment (kaiyue0329) [inline README.md:24]: The extension creates automatic `read-only` revisions for the user...

File: README.md [hunk 2]
```diff
@@ -3,26 +3,130 @@
...
+
+The following are some high-level features the extension provides:
+
+- ### Automatic Revisions
+
+  The extension creates automatic revisions for the user based on when files are saved in their workspace. These automatic revisions are tracked by the extension and users are capable or reviewing them directly in the application. 
+
+- ### View Revisions as a `Diff`
+
+  When reviewing a revision for an active editor, the revision will be displayed as a `diff` (lines added versus deleted) so that users are able to easily view the differences between the revision and the content of the current editor.
```
Comment (kaiyue0329) [inline README.md:28]: When reviewing a revision for an active editor, the `read-only` revision will be displayed as

File: README.md (hunk 1 above)
Comment (kaiyue0329) [inline README.md:24]: The extension creates automatic read-only revisions for the user...

File: README.md (hunk 2 above)
Comment (kaiyue0329) [inline README.md:28]: When reviewing a revision for an active editor, the read-only revision will be displayed as...

Comment (vince-fugnitto) [reply README.md:28]: This one I'm not sure about since we can have an inline diff which wouldn't apply.

**Comment by vince-fugnitto** [comment]
I placed the pull-request as 'ready for review', if you guys would like to improve it further we can do so as subsequent pull-requests (including `.gifs`).

**Comment by vince-fugnitto** [comment]
I'll hold off on this pull-request till we fix outstanding issues and cleanup.

**Pull request closed without merging**